packages/protocol/thread/thread.ts

```diff
@@ -1021,6 +1028,17 @@ class _ThreadFront {
     return null;
   }
 
+  getGeneratedLocation(locations: MappedLocation) {
+    const sourceIds = new Set<SourceId>(locations.map(location => location.sourceId));
+    return locations.find(location => {
+      const generated = this.getGeneratedSourceIds(location.sourceId);
+      if (!generated) {
+        return true;
+      }
+      return !generated.some(generatedId => sourceIds.has(generatedId));
```

Comment:
This feels like it could use a comment because I'm finding this check a bit hard to follow.